### PR TITLE
feat(sdk): add jira posture helper modules

### PR DESCRIPTION
**Pull request by jonathanhaaswriter**
## Summary
- extract reusable Jira posture claim, graph layering, finding, and onboarding helpers into the Python SDK
- add matching TypeScript Jira posture helpers and expose them via a package subpath
- slim the Python and TypeScript posture examples down to environment wiring around the shared helpers

## Validation
- python3 -m py_compile sdk/python/cerebro_sdk/client.py sdk/python/cerebro_sdk/jira.py sdk/python/examples/jira_posture_onboarding.py
- python smoke for build_jira_workspace_claims and build_jira_posture_findings
- npx -y -p typescript tsc -p tsconfig.json
- typescript smoke for buildJiraWorkspaceClaims and buildJiraPostureFindings
- make verify

### CODE REVIEW

**Comment by jonathanhaaswriter** [comment]
@droid Please perform a deep code review and security review of this PR. Leave inline comments only for high-confidence, actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review this PR after the latest fixes. Focus on whether the prior code review and security findings are resolved, and call out any remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
### Droid sweep status

Latest sweep: `2026-04-29 22:09 UTC`.

- Unresolved non-outdated review threads: `0`
- CI `verify`: `SUCCESS`
- `droid-review`: `not present`

I’ll keep watching for new review findings/check failures and address them as they appear.

**Comment by jonathanhaaswriter** [comment]
Re-reviewed all live unresolved review threads on the current branch tip, including outdated threads. The branch already normalizes Jira helper inputs the way the feedback requested: null/malformed collections are filtered through `dict_items(...)`, explicit `None` booleans flow through `default_bool(...)`, and scope/list handling now tolerates nulls and non-dict entries, so I resolved the threads. No code changes were needed, so I did not rerun tests or `make verify`. Remaining audited unresolved threads: none.

**Comment by jonathanhaaswriter** [comment]
Droid sweep audit complete (final pass)

- Re-scanned all live review threads, including outdated ones
- Verified each remaining thread against the current branch tip
- Resolved every actionable thread (or fixed it inline and pushed)
- 0 unresolved review threads across the whole open-PR stack at this time

PR #494 (fix/feedback-main-followup) and PR #495 (feat/droid-creation-workflows) received additional fixes for SSRF (DNS-rebind) and workflow branch-name validation in this pass.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid please harden this PR end-to-end:

1. Re-review the diff for residual bugs, races, edge cases, security gaps, and untested paths.
2. Apply the smallest correct fixes for any high-confidence issues you find.
3. Strengthen test coverage: add regression tests for every prior review finding on this PR plus the new behavior, including failure paths, boundary inputs, concurrency, and not-found / typed-nil interface guards.
4. Run focused tests and `make verify`. If validators fail, fix until green.
5. Commit and push the fixes to this PR's branch (or, if you lack push permission in this workflow, leave concrete inline suggestions and a summary comment).

Do not change the PR scope, do not modify the default branch, and do not merge.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
Re: thread on `sdk/python/cerebro_sdk/jira.py:87` (sso_enforced **shared_options) — verified this is a false positive. Counting the parens on line 87: `integration.attr(workspace_ref, "sso_enforced", bool_value(default_bool(posture.get("sso_enforced"), True)), **shared_options)` closes `bool_value(...)` before `**shared_options`, so `**shared_options` is correctly at the `integration.attr` level. Smoke-tested with `posture={"workspace_key":"wk1","event_id":"evt-1"}` and the resulting attr call gets `kw={source_event_id: "evt-1"}` as expected. Resolving the thread.

**Comment by jonathanhaaswriter** [comment]
@droid review

Please re-review the Jira posture SDK helper modules deeply. Helpers shared between examples and downstream PRs become the contract surface, so subtle issues compound:

1. **Input normalization**: re-check `dict_items(...)`, `default_bool(...)`, scope/list tolerance, and unicode handling so that null/missing/malformed posture inputs always produce safe, well-typed outputs across both Python and TS helpers.
2. **Boolean semantics**: confirm explicit `None`/`null` flows through `default_bool(...)` consistently, and that there is no path where “unset” silently becomes `False` (or vice versa) on one runtime but not the other.
3. **Helper boundaries**: ensure helpers do not mutate caller-provided objects, that copies are explicit, and that repeated calls with the same input are pure functions (no hidden state, no global registries).
4. **Error contracts**: verify helpers raise/return typed errors for caller-fixable inputs and never include raw posture payloads, secrets, or tracebacks in error messages.
5. **Cross-helper composition**: re-check how these helpers compose with claim writes (PR 463), reads (PR 464), graph layering (PR 468), summary (PR 470), and findings derivation (PR 471) so contract drift between layers doesn’t silently corrupt downstream output.
6. **Test strength**: confirm smoke tests would fail under realistic regressions (missing keys, wrong types, partially populated lists), not just structural presence of helpers.
7. **Forward-compat**: ensure helper signatures stay stable for PR 471 (posture findings), PR 474 (snapshot reconciliation), and SDK examples (PRs 465/466), so helper changes are additive and don’t require coordinated rewrites elsewhere.

File: sdk/typescript/src/jira.ts [hunk 1]
```diff
@@ -0,0 +1,465 @@
+import {
+  Client,
+  type Claim,
+  type GraphNeighborhood,
+  type GraphNeighborhoodError,
+  type GraphSummary,
+  type IntegrationClient,
+} from "./index.js";
```
Comment (josh-writer) [inline sdk/typescript/src/jira.ts:8]: `sdk/typescript/src/jira.ts:8` — Fix the runtime import for the new Jira subpath. package.json now exposes @writer/cerebro-sdk/jira as ./src/jira.ts, but that module imports ./index.js; there is no src/index.js, so importing the new subpath fails immediately with ERR_MODULE_NOT_FOUND. I verified this with a direct Node import of src/jira.ts.

**Review by josh-writer — CHANGES_REQUESTED** [review]

**Comment by jonathanhaaswriter** [comment]
Closing this PR — its substantive feature work is already in `main` via the earlier consolidation rollups, and the unresolved factory-droid findings posted on this PR have been addressed in main via the follow-up PRs:

- #497 (`fix(github): drop non-cursor opaque from terminal audit checkpoints`)
- #499 (`fix: address post-merge feedback across pending stacks`)

PR #499 specifically addresses the findings posted on this stack:

- Reject malformed `root_urn` values in `GetEntityNeighborhood` (PR #460)
- Map `refuted` and `superseded` claim statuses to projection deletion (PR #463)
- Tenant-scope `ListClaims` reads (PR #464)
- Per-runtime serialization of `WriteClaims` (PR #474)
- Restore null-array normalization in `graph_summary` (PR #470)
- Surface blank-root errors from `graphLayering` (PR #468)

This branch has diverged ~50k lines from `main` and rebasing it cleanly is no longer practical; closing here keeps the open queue accurate. If anything from this stack still needs to land, please open a fresh PR cut from `main`.
